Source/SwiftLintFramework/Models/LinterCache.swift

```diff
@@ -65,31 +70,47 @@ public final class LinterCache {
 
         let configurationDescription = configuration.cacheDescription
 
-        lock.lock()
-
-        guard let filesCache = cache[configurationDescription] as? [String: Any],
+        guard let filesCache = readCache[configurationDescription] as? [String: Any],
             let entry = filesCache[file] as? [String: Any],
             let cacheLastModification = entry[Key.lastModification.rawValue] as? TimeInterval,
             cacheLastModification == lastModification.timeIntervalSinceReferenceDate,
             let violations = entry[Key.violations.rawValue] as? [[String: Any]] else {
-                lock.unlock()
                 return nil
         }
 
-        lock.unlock()
         return violations.flatMap { StyleViolation.from(cache: $0, file: file) }
     }
 
     public func save() throws {
         guard let url = location else {
             throw LinterCacheError.noLocation
         }
+        guard !writeCache.isEmpty else {
+            return
+        }
+        flushWriteCache()
         lock.lock()
-        let json = toJSON(cache)
+        let json = toJSON(readCache)
         lock.unlock()
         try json.write(to: url, atomically: true, encoding: .utf8)
     }
 
+    internal func flushWriteCache() {
```

Comment:
Sounds good to me - this method shouldn't be used often, so I'm cool with it being more verbose